tests/light/functional_tests/source_drivers/python_fetcher/test_python_fetcher_custom_options.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2023 Attila Szakacs
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+
+def test_python_fetcher_custom_options(config, syslog_ng):
```

Comment:
We wouldn't need to test all python components for all types, if we just merged the implementation into one. Then a simple happy path that the option stuff is working would be enough in my opinion.

Test execution time is a scarce resource too.

Reply:
I would really like to test all of these, as even with the new `PythonOptions` class there are still some code duplication. Also the python confgen has a completely different implementation.

I added python confgen tests, and merged all of these together to one light TC, which runs for 0.22 seconds, so I guess it is good enough.